authorizations/message-filter/src/contract.rs

```diff
@@ -0,0 +1,408 @@
+use cosmwasm_std::{entry_point, to_binary, Addr, Uint128};
+use cosmwasm_std::{Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+use cw_auth_middleware::msg::IsAuthorizedResponse;
+use schemars::{JsonSchema, Map};
+use serde_derive::{Deserialize, Serialize};
+use serde_json_wasm::{from_str, to_string};
+
+use crate::error::ContractError;
+use crate::msg::{AuthorizationsResponse, ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{Authorization, Config, Kind, ALLOWED, CONFIG};
+use cw_auth_middleware::ContractError as AuthorizationError;
+
+const CONTRACT_NAME: &str = "crates.io:whitelist";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    U128(Uint128),
+    Number(u64), // Why doesn't u128 work here?
+    String(String),
+    Array(Vec<Value>),
+    Object(Map<String, Value>),
+}
+
+fn msg_to_value(msg: &CosmosMsg) -> Result<Value, ContractError> {
+    let serialized = to_string(msg).map_err(|_| ContractError::CustomError {
+        val: "invalid CosmosMsg".to_string(),
+    })?;
+
+    str_to_value(&serialized)
+}
+
+fn str_to_value(msg: &str) -> Result<Value, ContractError> {
+    from_str(msg).map_err(|_| ContractError::CustomError {
+        val: "invalid str".to_string(),
+    })
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
```

Comment:
I don't think this need to be a contract but a library imported directly by thing using it.

Reply:
It makes sense to refactor the message parsing and matching into a library. We would still need a message filter contract if we want DAOs to add auths in a generic way (for example: authorize an account to mint NFTs from the DAO account to give to members)